common/djangoapps/microsite_configuration/microsite.py

```diff
@@ -126,6 +126,7 @@ def _set_current_microsite(microsite_config_key, subdomain, domain):
     """
     config = settings.MICROSITE_CONFIGURATION[microsite_config_key].copy()
     config['subdomain'] = subdomain
+    config['microsite_config_key'] = microsite_config_key
```

Comment:
we are getting the microsite_key from lms Enviornment. _set_current_microsite called from a loop. 

 for key, value in settings.MICROSITE_CONFIGURATION.items():
        subdomain = value.get('domain_prefix')
        if subdomain and domain.startswith(subdomain):
            _set_current_microsite(key, subdomain, domain)
            return